### PR TITLE
James 2045 Correct dateTime Tests

### PR DESCRIPTION
**Pull request by chibenwa**
Fixes https://github.com/linagora/james-mime4j/pull/7

(1 approuval)

### CODE REVIEW

File: dom/src/main/javacc/org/apache/james/mime4j/field/datetime/DateTimeParser.jj [hunk 1]
```diff
@@ -225,7 +225,15 @@ int month() :
 String year() :
 {Token t;}
 {
-    t=<DIGITS> { return t.image; }
+    t=<DIGITS> 
+    {
+      String year = (String) t.image;
+      if (year.length() == 2) 
+      {
+        return "20" + year;
```
Comment (quynhn) [inline dom/src/main/javacc/org/apache/james/mime4j/field/datetime/DateTimeParser.jj:233]: I grew up from 19xx then at that time, we wrote "1978" by "78" also. But yes, from now, we can ignore the email from previous century :)

File: dom/src/test/java/org/apache/james/mime4j/field/datetime/DateTimeTest.java [hunk 2]
```diff
@@ -101,10 +103,9 @@ public void testNormalDate() throws ParseException {
 
     private void ensureAllEqual(String[] dateStrings) throws ParseException {
         for (int i = 0; i < dateStrings.length - 1; i++) {
-            Assert.assertEquals(
-                    new DateTimeParser(new StringReader(dateStrings[i])).parseAll().getDate().getTime(),
-                    new DateTimeParser(new StringReader(dateStrings[i + 1])).parseAll().getDate().getTime()
-            );
+            assertThat(new DateTimeParser(new StringReader(dateStrings[i])).parseAll().getDate().getTime())
```
Comment (quynhn) [inline dom/src/test/java/org/apache/james/mime4j/field/datetime/DateTimeTest.java]: It's quite difficult for me to understand what is equals to what. Could we separate realValue and expectedValue?

File: dom/src/test/java/org/apache/james/mime4j/field/datetime/DateTimeTest.java [hunk 3]
```diff
@@ -44,12 +44,12 @@ public void testNormalDate() throws ParseException {
 
 
         ensureAllEqual(new String[]{
-                "Fri, 21 Nov 1997 09:55:06 -0600", // baseline
+                "Fri, 21 Nov 2097 09:55:06 -0600", // baseline
```
Comment (quynhn) [inline dom/src/test/java/org/apache/james/mime4j/field/datetime/DateTimeTest.java:47]: Poor "Yahoo" people who used James since 19xx

**Review by quynhn — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**